src/js/api/auth/login.test.js

```diff
@@ -0,0 +1,67 @@
+import { login } from "./login";
+
+// Found at https://stackoverflow.com/questions/11485420/how-to-mock-localstorage-in-javascript-unit-tests
+// Used to avoid error due to localStorage not being defined
+class MockStorage {
+    constructor () {
+      this.storage = new Map()
+    }
+    setItem (key, value) {
+      this.storage.set(key, value)
+    }
+    getItem (key) {
+      return this.storage.get(key)
+    }
+    removeItem (key) {
+      this.storage.delete(key)
+    }
+    clear () {
+      this.constructor()
+    }
+}
+
+const badUserName = "oscaræ@norf.no";
+const badPassword = "badPasssa";
+
+const goodUserName = "oscar@noroff.no";
+const goodPassword = "buttfuck";
+
+const verifiedUser = {
+    name: "Oscar",
+    email: "oscar@noroff.no",
+    accessToken: "dkfsohosdih768736723**23nnl"
+}
+
+const fetchSuccess = () => {
+    return Promise.resolve({
+        ok: true,
+        status: 200,
+        statusText: "OK",
+        json: () => Promise.resolve({...verifiedUser})
+    })
+}
+
+const fetchFailure = () => {
+    return Promise.resolve({
+        ok: false,
+        status: 401,
+        statusText: "Unauthorized"
+    });
+}
+
+describe("login", () => {
+
+    it("returns an oject that contains an access token", async () => {
+        global.fetch = jest.fn (()=>fetchSuccess());
+        global.localStorage = new MockStorage;
+        const data = await login(goodUserName, goodPassword);
+        expect(data.accessToken).toEqual(verifiedUser.accessToken);
```

Comment:
@olee2 This test is failing. See the screenshot for details of the error
![ole](https://user-images.githubusercontent.com/25647480/204231497-54ccb597-25ec-4ba9-8479-e8b34f825360.PNG)

Reply:
The exact wording in the CA for the test in question is: `The login function returns a valid token when provided with valid credentials` The test fails because the function does not return a valid token when provided with valid credentials. It specifically deletes the token before returning the rest of the profile object. It is also not a requirement in the CA to have all tests passing. Please reconsider.